syntax = "proto3";

service UserService{
  rpc checkUser(UserCheckMessage) returns (UserResponse);
  rpc sendInterests(UserInterestsMessage) returns (UserResponse);
  rpc getInfoByAirport(UserInterestsMessage) returns (FlightsInfo);
}


message UserCheckMessage{
  string email = 1;
}

message UserInterestsMessage{
  string email = 1;
  string airport_code = 2;
  bool mode = 3;
}


message FlightInfo{
  string code_flight = 1;
  string time_departure = 2;
  string time_arrive = 3;
  string code_airport_departure = 4;
  string code_airport_arrive = 5;
}

message FlightsInfo{
  int32 status = 1;
  repeated FlightInfo info = 2;
}

message UserResponse {
  int32 status = 1;
  string message = 2; // Status code: 0 = success, 1 = error
}